packages/aws-cdk/lib/api/environment-resources.ts

```diff
@@ -108,6 +109,8 @@ export class EnvironmentResources {
       if (defExpectedVersion > version) {
         throw new Error(`This CDK deployment requires bootstrap stack version '${expectedVersion}', found '${version}'. Please run 'cdk bootstrap'.`);
       }
+      const notices = Notices.get();
+      notices.enqueuePrint(notices.forBootstrapVersion({ bootstrapVersion: version }));
```

Comment:
We enqueue here and not print to preserve printing all notices at the end of the command execution.